plugins/core-plugin/src/main/java/com/google/cloud/teleport/plugin/DockerfileGenerator.java

```diff
@@ -176,15 +172,12 @@ public Builder setEntryPoint(String[] entryPoint) {
       return this;
     }
 
-    public Builder setFilesToCopy(Map<String, Set<String>> filesToCopy) {
-      StringBuilder files = new StringBuilder();
-      for (String file : filesToCopy.keySet()) {
-        files.append(String.format("COPY %s", file));
-        if (!filesToCopy.get(file).isEmpty()) {
-          files.append(String.format(" %s", String.join(" ", filesToCopy.get(file))));
-        }
-        files.append(" /$WORKDIR/\n");
+    public Builder setFilesToCopy(List<String> filesToCopy) {
+      StringBuilder files = new StringBuilder("COPY");
+      for (String file : filesToCopy) {
+        files.append(String.format(" %s", file));
       }
+      files.append(" $WORKDIR/\n");
```

Comment:
Why do we need this change? Doesn't this make this function less robust since it will now leave behind `COPY $WORKDIR` if there are no filesToCopy?

Reply:
(to be clear, I think switching to a list input is a good idea, I just don't know why we're changing the core ordering of the logic.

Reply:
I made an assumption the list would never be empty. I added preconditions checks (to this method and others) to reinforce the assumptions.

I also added Java docs to all the methods to make the usage clearer

Reply:
I'm ok with this and am not going to block on it, but I do actually like the original version more; in general it is better to be permissive with your inputs if you can (https://en.wikipedia.org/wiki/Robustness_principle)